plugins/modules/rma_workflow_manager.py

```diff
@@ -865,16 +867,48 @@ def unmark_device_for_replacement(self):
                 "Error while unmarking device for replacement"
             )
             self.status = task_result["status"]
-            self.msg = task_result["msg"]
+            self.msg = "RMA failed to replace the device: {0}".format(task_result["msg"])
             if self.status == "success":
                 self.result['changed'] = True
 
-        except Exception as e:
+        except Exception:
             self.status = "failed"
-            self.msg = "Exception occurred while unmarking device for replacement: {0}".format(str(e))
+            self.msg = "RMA failed to replace the device: No device found for unmarking replacement"
             self.log(self.msg, "ERROR")
         return self
 
+    def return_replacement_devices(self):
+        """
+        Retrieves the ID of the device ready for replacement in Cisco Catalyst Center.
+
+        Parameters:
+            - self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+
+        Returns:
+            - device_id (str or None): The ID of the device that is ready for replacement, or None if no such device is found.
+
+        Description:
+            This method retrieves the ID of a device that is marked as "READY-FOR-REPLACEMENT" from Cisco Catalyst Center.
```

Comment:
From the code I understand that  you are traversing the list of devices..

        This method fetches a list of devices with their replacement status from Cisco Catalyst Center.
        It then checks for the first device with a "READY-FOR-REPLACEMENT" status and returns its ID.
        The method exits early if such a device is found.

Reply:
update the code accordingly